### PR TITLE
Update cluster-api to v1.4.1

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
**What this PR does / why we need it**:
- Update cluster-api to v1.4.1

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
Update cluster-api to v1.4.1
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,7 +6,7 @@ go 1.19
 replace (
 	github.com/onsi/ginkgo/v2 => github.com/onsi/ginkgo/v2 v2.9.1
 	github.com/onsi/gomega => github.com/onsi/gomega v1.27.4
```
Comment (mkumatag) [inline go.mod:8]: wondering if these two also need update with that?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (Prajyot-Parab) [reply go.mod:8]: Yes, `1.4.1 capi` uses those versions so should be fine.

Comment (Prajyot-Parab) [reply go.mod]: Done.

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**